### DISCUSSION TITLE
unable to load images from backend

### DISCUSSION THREAD

**Discussion by vinay-reddy-306** [niklasvh/html2canvas#2682] [Q&A]
Hi, I was trying to convert html to image, the result seems like the images which we were getting urls from backend got missing.
Here is the reference code that I have tried with:

var imgContainer = document.getElementById("student-chart");
html2canvas(imgContainer,{allowTaint : true, useCORS : true}).then(function(canvas) {
        var link = document.createElement("a");
        document.body.appendChild(link);
        link.download = "xxxxxx.png";
        link.href = canvas.toDataURL("image/png");
        link.target = '_blank';
        link.click();
})
can anyone please share your inputs on resolving this issue? Thank you

**Comment by komitoff** [comment]
Hey I see not much activity on this topic. 
I am trying to achieve basically the same. I am working on a web app using vaadin 14 as a UI framework. My idea is that the user has the option to capture the screen.
Here is my code: 

html2canvas(document.getElementsByTagName('vaadin-app-layout')[0], {})
                .then(canvas => {
      		let a = document.createElement("a");
      		document.body.appendChild(a);
      		a.href = canvas.toDataURL(`image/jpg`, 1.0);
      		a.download = "screenshot.jpg";
      		a.click();
              a.remove();
          })
          
  If I run the code like this I get a screenshot downloaded, but images are missing (if there are images on the screen). I read here and there and if I apply some options to the html2canvas function (useCORS:true), I see that my browser log is full of errors basically telling me that:
  `No 'Access-Control-Allow-Origin' header is present on the requested resource.` 
 It is written in the documentation of this library that we can provide a proxy URL that will help solving this issue. But since I am using spring as a backend and I was able to find some exaples with ready2use libraries for nodejs and php, can somebody explain a bit further, what exactly the PROXY should do ? How the proxy end-point is called and what type of parameters is expected to get and return in order to make it work. I assume that the proxy should handle and download the images that are blocked because of the CORS policy, but couldn't find any good information on how it should do it. 
 Thanks in advance!
